### PR TITLE
init commit of interoute oem files

### PR DESCRIPTION
**Pull request by xied75**
This is the initial PR for interoute OEM build of CoreOS. It mainly combines cloudstack related files with vmtoolsd since we are based on VMware hypervisor. We also modified the way we expose network information into the environment file to make utilising our MPLS VPN network easier.

### CODE REVIEW

File: coreos-base/oem-interoute/oem-interoute-0.0.2.ebuild [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	dobin "${FILESDIR}/cloudstack-set-guest-password"
+	dobin "${FILESDIR}/cloudstack-ssh-key"
+	dobin "${FILESDIR}/cloudstack-coreos-cloudinit"
+	dobin "${FILESDIR}/coreos-setup-environment"
+
+	insinto "/usr/share/oem"
+	doins "${T}/cloud-config.yml"
+	doins "${FILESDIR}/oem-release"
+
+        insinto "/etc/systemd/system"
```
Comment (marineam) [inline coreos-base/oem-interoute/oem-interoute-0.0.2.ebuild]: OEM's should not install to /etc or anywhere else outside of /usr/share/oem. This service file needs to be embedded into your cloud-config.yml.

File: coreos-base/oem-interoute/files/cloud-config.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+          [Service]
+          Type=oneshot
+          EnvironmentFile=/etc/environment
+          ExecStart=/usr/share/oem/bin/cloudstack-coreos-cloudinit
+    oem:
+      id: interoute
+      name: Interoute 
+      version-id: @@OEM_VERSION_ID@@
+      home-url: http://interoute.com/
+      bug-report-url: https://github.com/coreos/coreos-overlay
```
Comment (marineam) [inline coreos-base/oem-interoute/files/cloud-config.yml]: This should be `https://github.com/coreos/bugs/issues` these days. If other OEMs list this url we should fix that.

File: coreos-base/oem-interoute/files/cloud-config.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      - name: vmtoolsd.service
+        command: start
+        content: |
+          [Unit]
+          Description=VMware Tools Agent
+          Documentation=http://open-vm-tools.sourceforge.net/
+          ConditionVirtualization=vmware
+
+          [Service]
+          ExecStartPre=/usr/bin/ln -sfT /usr/share/oem/vmware-tools /etc/vmware-tools
```
Comment (crawford) [inline coreos-base/oem-interoute/files/cloud-config.yml]: Why is this here?

Comment (marineam) [inline coreos-base/oem-interoute/files/cloud-config.yml]: vmware-tools needs that path and cloudinit runs too late to do this with tmpfiles, iirc it is this way in the existing vmware OEM

Comment (xied75) [inline coreos-base/oem-interoute/files/cloud-config.yml]: Our hypervisor is VMware. We need the scaling stuff.

-----Original Message-----

Comment (xied75) [inline coreos-base/oem-interoute/files/cloud-config.yml]: I thought I copied from it, no?

-----Original Message-----

File: coreos-base/oem-interoute/files/cloud-config.yml (hunk 2 above)
Comment (marineam) [inline coreos-base/oem-interoute/files/cloud-config.yml]: This still need to be updated

**Comment by marineam** [comment]
LGTM, any thoughts @crawford?

**Comment by crawford** [comment]
LGTM
